internal/assert/assertion_mongo.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+// assertion_mongo.go contains MongoDB-specific extensions to the "assert"
+// package.
+
+package assert
+
+import (
+	"fmt"
+	"reflect"
+	"unsafe"
+)
+
+// DifferentAddressRanges asserts that two byte slices reference distinct memory
+// address ranges, meaning they reference different underlying byte arrays.
+func DifferentAddressRanges(t TestingT, a, b []byte) {
```

Comment:
Do you think it would be worth adding a "Helper()" function to the "TestingT" interface?

Reply:
Calling `t.Helper()` is a good idea. I can actually follow the same pattern that the `assert` package uses [here](https://github.com/mongodb/mongo-go-driver/blob/e748329f8e9b349010d3f185592c76ee22719b63/internal/assert/assertions.go#L293-L295).